tools/publish-crates.sh

```diff
@@ -4,41 +4,78 @@ set -euo pipefail
 cd "$(dirname "$0")/.."
 
 DRY_RUN=0
+ALLOW_DIRTY=0
+NEW_CRATE_OWNERS=("tyler@clockworklabs.io" "zeke@clockworklabs.io")
 
-if [ "$#" != "0" ]; then
-    if [ "$1" != "--dry-run" ]; then
-        echo "$1 is not a valid flag";
-        exit 1;
-    else
-        DRY_RUN=1
-    fi
-fi
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --dry-run)
+            DRY_RUN=1
+            ;;
+        --allow-dirty)
+            ALLOW_DIRTY=1
+            ;;
+        *)
+            echo "Invalid argument: $1"
+            exit 1
+            ;;
+    esac
+    shift
+done
 
-if [ $DRY_RUN != 1 ] ; then
-	echo "You are about to publish to crates.io (dry run is false.)"
-	echo "We are also going to do a test install after publishing. This will remove any version of spacetimedb-cli you may have installed."
-	echo
-	echo "Press [Enter] to continue."
-	read -r
+if [ $DRY_RUN -ne 1 ]; then
+    echo "You are about to publish to crates.io (dry run is false.)"
+    echo "We are also going to do a test install after publishing. This will remove any version of spacetimedb-cli you may have installed."
+    echo
+    echo "Press [Enter] to continue."
+    read -r
 fi
 
 BASEDIR=$(pwd)
-declare -a CRATES=("metrics" "primitives" "sql-parser" "bindings-macro" "bindings-sys" "data-structures" "sats" "lib" "schema" "bindings" "table" "vm" "client-api-messages" "paths" "commitlog" "durability" "fs-utils" "snapshot" "expr" "core" "client-api" "standalone" "cli" "sdk")
+declare -a CRATES=("metrics" "primitives" "sql-parser" "bindings-macro" "bindings-sys" "data-structures" "sats" "lib" "schema" "bindings" "table" "vm" "client-api-messages" "paths" "commitlog" "durability" "fs-utils" "snapshot" "expr" "execution" "physical-plan" "query" "core" "client-api" "standalone" "cli" "sdk")
 
-for crate in "${CRATES[@]}" ; do
-	if [ ! -d "${BASEDIR}/crates/${crate}" ] ; then
-		echo "This crate does not exist: ${crate}"
-		exit 1
-	fi
+for crate in "${CRATES[@]}"; do
+    if [ ! -d "${BASEDIR}/crates/${crate}" ]; then
+        echo "This crate does not exist: ${crate}"
+        exit 1
+    fi
 done
 
-for crate in "${CRATES[@]}" ; do
-	cd "${BASEDIR}/crates/${crate}"
-	if [ $DRY_RUN == 1 ] ; then
-		cargo publish --dry-run
-	else
-		cargo publish
-	fi
+for crate in "${CRATES[@]}"; do
+    cd "${BASEDIR}/crates/${crate}"
+
+    PUBLISH_CMD="cargo publish"
+    [[ $DRY_RUN -eq 1 ]] && PUBLISH_CMD+=" --dry-run"
+    [[ $ALLOW_DIRTY -eq 1 ]] && PUBLISH_CMD+=" --allow-dirty"
+
+    # Check if crate exists on crates.io
+    if ! cargo search "$crate" --limit 1 | grep -q "^$crate ="; then
```

Comment:
I'm not sure that this `--limit 1` is helping us if we're already `grep`ing. Does `cargo search` guarantee that an exact string match would appear first?

Here's my canonical "cargo search is weird" example:
```
$ cargo search asdfasdf
lisho = "0.1.5"                 # A simple personal link shortener with no external dependencies in under 200 lines of Rust.
okapi-response-mac = "0.1.1"    # A macro for creating response code based on responses structs rather than typing and implementi…
umami_metrics = "0.1.0"         # A rust library to send metrics data to the Umami metrics platform
```

```suggestion
    if ! cargo search "$crate" | grep -q "^$crate ="; then
```